### PR TITLE
Create DAO for review table

### LINKED ISSUE #4: [SUBTASK] Add new DAO for Reviews
**What feature is this a subtask of**
#3 

**What will be implemented in this subtask**
Modifying the backend to add a new Data Access Object (DAO) that interfaces with the new table of reviews. The methods need to be able to insert a new review record and retrieve each user in the table + their average review score (maybe separated by (user, document) keys instead of just user).
  
**What is the acceptance criteria of this subtask**
What defines when this feature is suitably complete?

**What files will be created/modified**
Create a new DAO file `ReviewDao.java` in the same folder as all the other DAO files. Add a new test file `TestReviewDao.java` in the same folder as `TestJpa.java`. 

**What dependencies does this issue have or create**
Dependent on <INSERT ISSUE FOR MAKING TABLE HERE>

**What is the estimated effort of this subtask**
Approximately 2 days

### PR DESCRIPTION
**Pull request by knassre-cmu**
## Issue
Resolves #4 

## Description
Create a Data-Access-Object (DAO) to access the Review database created in #7. Adds a method for adding a new entry to the table, and a query method to get the average review score for each username/docId combo. Also creates a DTO file so that the averages can be returned as list of DTO objects. Each DTO has only 3 fields: username, docId, average review score. Added a test file that inserts multiple entries and verifies that there averages are correctly queried.

## Files changed
`docs-core/src/main/java/com/sismics/docs/core/dao/ReviewDao.java`
`docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java`
`docs-core/src/main/java/com/sismics/docs/core/dao/dto/ReviewAggDto.java`
Edited `docs-core/src/main/resources/db/update/dbupdate-028-0.sql`

## Notes for reviewers
N/A

## Checklist
Fill out this checklist before merging the PR:
- [x] Created issue
   - [x] Issue has a relevant labels
   - [x] Issue added to the kanban board with its status as In-Progress
   - [x] Issue added to the relevant milestone
   - [x] Issue mentioned in the first section of this PR description
- [x] Testing
   - [x] Added new tests for the changes (or modified existing tests)
   - [x] Passed most recent round of CI
- [x] Review
   - [x] Received first round of review comments
   - [x] Addressed first round of review comments
   - [x] Received second round of review comments
   - [x] Addressed second round of review comments

### CODE REVIEW

File: docs-core/src/main/java/com/sismics/docs/core/dao/ReviewDao.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    public void create(
+        String userId, String username, String documentId, String reviewerId, Integer rating)
+        {
+        // Make a new review object and give it a random ID
+        Review review = new Review();
+        String reviewId = UUID.randomUUID().toString();
+        review.setReviewId(reviewId);
+
+        // Populate hte other fields
```
Comment (aruiz2) [inline docs-core/src/main/java/com/sismics/docs/core/dao/ReviewDao.java]: "the"

File: docs-core/src/main/java/com/sismics/docs/core/dao/dto/ReviewAggDto.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.sismics.docs.core.dao.dto;
+
```
Comment (aruiz2) [inline docs-core/src/main/java/com/sismics/docs/core/dao/dto/ReviewAggDto.java:2]: Add description for the public class?

File: docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                Assert.assertEquals(91.5, res.getAvgRating(), 0.1);
+            } else if (res.getUsername() == "alice" && res.getDocId() == "DocB") {
+                found2 = true;
+                Assert.assertEquals(91, res.getAvgRating(), 0.1);
+            } else if (res.getUsername() == "chris" && res.getDocId() == "DocC") {
+                found3 = true;
+                Assert.assertEquals(80, res.getAvgRating(), 0.1);
+            }
+        }
+
```
Comment (aruiz2) [inline docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java]: nit: remove line

File: docs-core/src/main/java/com/sismics/docs/core/dao/dto/ReviewAggDto.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+public class ReviewAggDto {
```
Comment (jding712) [inline docs-core/src/main/java/com/sismics/docs/core/dao/dto/ReviewAggDto.java:7]: Add a short description of ReviewAggDto

**Review by aruiz2 — COMMENTED** [review]
ADD SORTING!

**Review by jding712 — APPROVED** [review]
Looks good!

File: docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        reviewDao.create("1", "alice", "DocB", "RevB", 91);
+        reviewDao.create("2", "chris", "DocC", "RevA", 80);
+        reviewDao.create("1", "alice", "DocA", "RevB", 93);
+
+        // Verify that the lookup query contains these three in there
+        Boolean found1 = false;
+        Boolean found2 = false;
+        Boolean found3 = false;
+        List<ReviewAggDto> output = reviewDao.getAverageScorePerApplication();
+        for (ReviewAggDto res : output) {
```
Comment (aruiz2) [inline docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java]: rename "res" to something more readable

File: docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        reviewDao.create("1", "alice", "DocA", "RevA", 90);
+        reviewDao.create("1", "alice", "DocB", "RevB", 91);
+        reviewDao.create("2", "chris", "DocC", "RevA", 80);
+        reviewDao.create("1", "alice", "DocA", "RevB", 93);
+
+        // Verify that the lookup query contains these three in there
+        Boolean found1 = false;
+        Boolean found2 = false;
+        Boolean found3 = false;
+        List<ReviewAggDto> output = reviewDao.getAverageScorePerApplication();
```
Comment (aruiz2) [inline docs-core/src/test/java/com/sismics/docs/core/dao/jpa/TestReviewDao.java]: rename "output" to something more readable

**Comment by knassre-cmu** [comment]
@aruiz2 Revisions made. Does this look good enough for approval?

**Review by aruiz2 — APPROVED** [review]
Looks good to me! Nice job!

**Pull request merged**